filament/backend/src/vulkan/VulkanPipelineCache.cpp

```diff
@@ -181,8 +182,9 @@ void VulkanPipelineCache::getOrCreateDescriptors(
         }
     } else {
         for (uint32_t i = 0; i < DESCRIPTOR_TYPE_COUNT; ++i) {
-            descriptorSets[i] = mDescriptorSetArena[i].back();
-            mDescriptorSetArena[i].pop_back();
+            auto search = mDescriptorSetArena[i].find(layout->pipelineLayout);
```

Comment:
Are we 100% sure that this will never return `end()`?  If so, let's add a code comment that explains why.

Reply:
It's replaced to vector.